### DISCUSSION TITLE
问一下各位，你们的openclash的meta内核能正常用吗？

### DISCUSSION THREAD

**Discussion by xiandu999-yu** [vernesong/OpenClash#3846] [Q&A] [answered]
我的不知道为啥，连接那栏里面只有IP没有域名[油管看不了]...
![image](https://github.com/vernesong/OpenClash/assets/58256481/82d263dc-26e3-4a3d-a290-529b9887adcf)
![image](https://github.com/vernesong/OpenClash/assets/58256481/46f1404f-057a-40ec-9b55-fd6314fd395f)

**Comment by Plutonium141** [comment]
dns没处理好 排查从客户端到openwrt的dns设置，确保局域网的dns流量能到clash的7874端口，如果不想弄dns 打开域名嗅探也能缓解这个问题

**Reply by xiandu999-yu** [reply]
DNS我用的是SmartDNS...而且同样配置我原版clash_tun内核就能有域名的好吗？虽然clash_tun内核也有问题[socks转发不了UDP]

**Reply by xiandu999-yu** [reply]
![image](https://github.com/vernesong/OpenClash/assets/58256481/70a94b56-02ff-444c-a560-52a14bdf30d1)
这嗅探没用，缓解不了，只能开劫持
![image](https://github.com/vernesong/OpenClash/assets/58256481/8be96efb-8c5a-437a-9cda-e613a4067e43)
![image](https://github.com/vernesong/OpenClash/assets/58256481/6de4d0cc-8593-4112-b5b4-9619cd6e6f0b)

**Reply by xiandu999-yu** [reply]
哦豁，手机WiFi填软路由IP为网关也出现不劫持下差不多一模一样的状况[至少谷歌上不去，面板里连接一栏里域名缺胳膊少腿的]。。。

**Comment by Plutonium141** [comment]
> ![image](https://private-user-images.githubusercontent.com/58256481/322894080-6de4d0cc-8593-4112-b5b4-9619cd6e6f0b.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTMzMjM1MjYsIm5iZiI6MTcxMzMyMzIyNiwicGF0aCI6Ii81ODI1NjQ4MS8zMjI4OTQwODAtNmRlNGQwY2MtODU5My00MTEyLWI1YjQtOTYxOWNkNmU2ZjBiLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA0MTclMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNDE3VDAzMDcwNlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWMzOGI5NTkyNTc1ZDY2ODRiOTM1NDdlNTVkNDhhNGEyZTY2MjA4NmFiMDRhMmY2NDMxMWI3MmZiZWFlMTBkMGQmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.m_omEEyVfu_yurD65awmCRb2VeJWhRKckCOrS2VnPiw)

这图是开了嗅探的效果还是开劫持的效果？

**Reply by xiandu999-yu** [reply]
???

**Reply by xiandu999-yu** [reply]
你图呢？

**Reply by Plutonium141** [reply]
图就是你发的那张连接的截图

**Reply by xiandu999-yu** [reply]
开了嗅探，没用...

**Comment by Plutonium141** [comment]
我这边的meta内核是正常的，嗅探域名那一列有值就算嗅探有用
用smartdns一般是在clash里配置一条nameserver指向smartdns，然后在openclash开启劫持或者是手动配置dnsmasq的dns转发到7874，smartdns那边不做任何劫持

**Reply by xiandu999-yu** [reply]
其实只要DNS劫持开了就行，选防火墙转发SmartDNS也不用改设置...

**Comment by xiaoyangdkj** [comment]
在启用“探测纯ip连接”的同时启用“自定义流量探测设置”，随后在下面随便加一条域名
![Screenshot_20240421_200341_com.android.chrome_edit_697802700456023.jpg](https://github.com/vernesong/OpenClash/assets/82305021/39b86bbf-1c95-4eda-8a60-0069c98dd384)

**Reply by xiandu999-yu** [reply, marked as answer]
不用了，我已经发现问题了，是一定要开DNS劫持才行...
